anitya/db/models.py

```diff
@@ -496,6 +523,24 @@ def all(cls, session, page=None, count=False):
         else:
             return query.all()
 
+    @classmethod
+    def ready_for_check(cls, session, time):
```

Comment:
Like the above function, this doesn't belong on the model class (even though there's lot of functions like this). Instead, if it's used in multiple places, it should be put into a SQLAlchemy Query object that sub-classes https://github.com/release-monitoring/anitya/blob/7d115fca74a424e55e1b17743aae3e7039d59b6c/anitya/db/meta.py#L104

Since this is used in just one place, I'd just write the query out in the cron job.

Also, just a note (and I know this only here because existing code does it), but you don't need to pass the session around like this. SQLAlchemy creates sessions that are local to threads: https://github.com/release-monitoring/anitya/blob/7d115fca74a424e55e1b17743aae3e7039d59b6c/anitya/db/meta.py#L42 and you can get the current session by doing ``db.Session()``. The same session is also used when doing ``db.Project.query``.

Reply:
I just tried to make it more coherent with existing code. Didn't know, that this is bad practice.

Reply:
I have one question regarding this. Where should I place methods, that are relevant to project like `check_project_release()`? Is this a good candidate for `models.db.Project` method?

Reply:
> I just tried to make it more coherent with existing code. Didn't know, that this is bad practice.

Yeah, no worries, I never got around to cleaning up the models.

> I have one question regarding this. Where should I place methods, that are relevant to project like check_project_release()? Is this a good candidate for models.db.Project method?

That's likely a good candidate. It deals with a single project and performs something other than changing the database row. The general rule of thumb that I follow is things that methods that deal with a single row live on the project model class, things that deal with queries and multiple rows live on the Query object.